src/sign/mod.rs

```diff
@@ -167,6 +188,17 @@ impl CoseSign {
             self.payload.as_ref().unwrap_or(&vec![]),
         )
     }
+
+    /// Construct the to-be-signed data for this object, using a detached payload.
+    fn tbs_detached_data(&self, payload: &[u8], aad: &[u8], sig: &CoseSignature) -> Vec<u8> {
```

Comment:
I think this should fail if `self.payload.is_some()`, to reduce the chances of users mixing things up.

Of course, that makes life more complicated because returning a `Result` here doesn't align with the error types elsewhere.  Maybe a documented `panic!` for now?

Reply:
Added a documented panic when self.payload.is_some().